pilot/pkg/model/endpointshards.go

```diff
@@ -288,44 +288,48 @@ func (e *EndpointIndex) UpdateServiceEndpoints(
 	newIstioEndpoints := istioEndpoints
 	if features.SendUnhealthyEndpoints.Load() {
 		oldIstioEndpoints := ep.Shards[shard]
-		newIstioEndpoints = make([]*IstioEndpoint, 0, len(istioEndpoints))
-
-		// Check if new Endpoints are ready to be pushed. This check
-		// will ensure that if a new pod comes with a non ready endpoint,
-		// we do not unnecessarily push that config to Envoy.
-		// Please note that address is not a unique key. So this may not accurately
-		// identify based on health status and push too many times - which is ok since its an optimization.
-		emap := make(map[string]*IstioEndpoint, len(oldIstioEndpoints))
-		nmap := make(map[string]*IstioEndpoint, len(newIstioEndpoints))
-		// Add new endpoints only if they are ever ready once to shards
-		// so that full push does not send them from shards.
-		for _, oie := range oldIstioEndpoints {
-			emap[oie.Address] = oie
-		}
-		for _, nie := range istioEndpoints {
-			nmap[nie.Address] = nie
-		}
 		needPush := false
-		for _, nie := range istioEndpoints {
-			if oie, exists := emap[nie.Address]; exists {
-				// If endpoint exists already, we should push if it's health status changes.
-				if oie.HealthStatus != nie.HealthStatus {
+		if oldIstioEndpoints == nil {
```

Comment:
When oldIstioEndpoints are null, given istiod is restarted, how is the execution not entering the below code branch.
```
	ep, created := e.GetOrCreateEndpointShard(hostname, namespace)
	// If we create a new endpoint shard, that means we have not seen the service earlier. We should do a full push.
	if created {
		log.Infof("Full push, new service %s/%s", namespace, hostname)
		pushType = FullPush
	}
```

Reply:
I see, without the change, PushType is geting updatd from FullPush to NoPush.

Would skipping the iteration through the endpoints when PushType is FullPush. 
But that would also make the codeflow different for the fresh istiod pod and the old istiod pod.